app_dart/test/request_handlers/get_branches_test.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:cocoon_service/src/request_handlers/get_branches.dart';
+import 'package:cocoon_service/src/request_handling/body.dart';
+import 'package:github/server.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../src/datastore/fake_cocoon_config.dart';
+import '../src/request_handling/fake_http.dart';
+import '../src/request_handling/request_handler_tester.dart';
+
+const String branchRegExp = '''
+      master
+      ^flutter-[0-9]+\.[0-9]+-candidate\.[0-9]+
+      ''';
+
+void main() {
+  group('GetBranches', () {
+    FakeConfig config;
+    FakeHttpClient branchHttpClient;
+    RequestHandlerTester tester;
+    GetBranches handler;
+    List<String> githubBranches;
+
+    Stream<Branch> branchStream() async* {
+      for (String branchName in githubBranches) {
+        final CommitDataUser author = CommitDataUser('a', 1, 'b');
+        final GitCommit gitCommit = GitCommit();
+        final CommitData commitData = CommitData('sha', gitCommit, 'test',
+            'test', 'test', author, author, <Map<String, dynamic>>[]);
+        final Branch branch = Branch(branchName, commitData);
+        yield branch;
+      }
+    }
+
+    setUp(() {
+      final MockGitHub github = MockGitHub();
+      final MockRepositoriesService repositories = MockRepositoriesService();
+
+      config = FakeConfig(githubClient: github);
+      branchHttpClient = FakeHttpClient();
+      tester = RequestHandlerTester();
+      handler = GetBranches(
+        config,
+        branchHttpClientProvider: () => branchHttpClient,
+        gitHubBackoffCalculatorBranch: (int attempt) => Duration.zero,
+      );
+
+      const RepositorySlug slug = RepositorySlug('flutter', 'flutter');
+      when(github.repositories).thenReturn(repositories);
+      when(repositories.listBranches(slug)).thenAnswer((Invocation _) {
+        return branchStream();
+      });
+    });
+
+    test('returns branches matching regExps', () async {
```

Comment:
There's a lot of different try/catch if/else logic blocks that I believe should be tested here. I think moving the loadRegExp to a separate util file would make it easier to focus on this logic for a test.

Reply:
moved to `utils.dart` and added corresponding tests.